### DISCUSSION TITLE
packageRules cannot combine both matchUpdateTypes and ignoreDeps.

### DISCUSSION THREAD

**Discussion by p4block** [renovatebot/renovate#30937] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, latest

### Please tell us more about your question or problem

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:recommended"
  ],
  "argocd": {
      "fileMatch": [
          "\\.yaml$"
      ]
  },
  "kubernetes": {
    "fileMatch": ["\\.yaml$"]
  },
  "dependencyDashboard": true,
  "packageRules": [
    {
      "automerge": true,
      "matchUpdateTypes": ["patch"],
      "matchCurrentVersion": "!/^0/",
      "ignoreDeps": ["flakey-app"],
      "ignoreTests": true
    }
  ]

}

``` 

I have some apps I don't want even patches automerged, but this configuration isn't allowed, how to achieve this? 
The opposite usecase, a explicit list of apps I want to automerge patches for, seems just as impossible right now.

**Comment by rarkins** [comment]
Please explain in plain english what you're trying to achieve with that rule

**Reply by p4block** [reply]
Renovate is creating MRs to update application versions in kubernetes yaml, argocd applications and helm values. 
Some of this MRs are for internal tools and I don't want to auto merge them, even if the MR is for a patch bump. I want patch bumps to be automerged for everything else.

I can also configure it the opposite way, I want to specify a list of app names I would like to automatically merge the patch MRs for. 

A third option would be to just ignore these apps altogether from the MR generator if that's possible

**Reply by rarkins** [reply, marked as answer]
"I want to automerge all patch updates, except if they're v0 updates, and except if it's `flakey-app`". Does that capture the goal of your rule exactly?

You may just need to switch from `ignoreDeps` (which is not supported in packageRules) to `"matchPackageNames": ["!flakey-app"]`

**Reply by p4block** [reply]
Thanks a lot, that was just what I needed. It actually got me to find out how renovate is actually working.

**Reply by p4block** [reply]
Due to some internal discussion I resorted to this config in the end, flipping to logic to simply not make PRs for the flakey app(s) at all and handpick which things I want to automerge 1 by 1. 
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:recommended",
    ":disableRateLimiting"
  ],
  "argocd": {
      "fileMatch": [
          "\\.yaml$"
      ]
  },
  "kubernetes": {
    "fileMatch": ["\\.yaml$"]
  },
  "dependencyDashboard": true,
  "packageRules": [
    {
      "description": "Packages that developers handle, don't even make MRs",
      "matchPackageNames": [
        "flakey-app",
        ],
      "enabled": false
    },
    {
      "description": "Packages to automerge patches for. Very rarely will cause breakage. Supersedes X.X.* versions in charts",
      "matchUpdateTypes": [ "patch" ],
      "matchCurrentVersion": "!/^0/",
      "automerge": true,
      "ignoreTests": true,
      "matchPackageNames": [
        "ingress-nginx",
        "gitlab-runner",
        "gitlab"
      ]
    },
    {
      "description": "Packages to automerge minors and patches for. Some stuff will break eventually, but at least it's at 6am",
      "matchUpdateTypes": [ "minor", "patch" ],
      "matchCurrentVersion": "!/^0/",
      "automerge": true,
      "ignoreTests": true,
      "matchPackageNames": [
        "argo-cd",
        "cert-manager",
        "external-secrets"
      ]
    },
  ]

}
```
